articles/key-design-of-hdfs-architecture/index.md

```diff
@@ -0,0 +1,120 @@
+
+***NameNode/DataNode in master/slave mode:*** HDFS consists of NameNodes and DataNodes that work in a master/slave architecture. A single cluster consists of only one NameNode which regulates data access by clients and manages the namespace within the file system.
+
+The DataNode receives instructions from the NameNode on when to Create, Delete and replicate data blocks.
+
+***Unified file system Namespace:*** HDFS is presented externally as a coherent file system.
+Any external process perceives the system as one unified system.
+
+***Data replication:*** In HDFS, a file’s blocks are replicated for fault tolerance and the number of replicas can be specified by an application. This can be done at creation time but is subject to change at will.
+
+***Metadata persistence:*** The HDFS NameNode stores the namespace. The NameNode consistently records every change that occurs in file system metadata in a transaction log file called the EditLog. Whenever a new file is created in HDFS, the NameNode inserts a record into the EditLog indicating the creation of the new file.
+
+This information is also synchronized between the Active and the Standby NameNode periodically.
+
+***Robustness:*** HDFS stores data reliably even when failures occur. Its robustness takes into account the 3 common types of failures: DataNode failure, NameNode failure, and network failures.
+
+***Data organization:*** Data is stored by blocks of size 64MB in HDFS.
+
+***HDFS Data Integrity Assurance:*** HDFS ensures the completeness of the stored data by implementing reliability processing in case of failure of each component. HDFS accomplishes this by doing the following;
+
+- Reconstructs data replicas in invalid data disks - the DataNode periodically reports blocks’ messages to the NameNode, if one replica (block) is failed, the NameNode will start the procedure to recover lost replicas
+- Ensures data balance among DataNodes - the HDFS architecture is configured with the data balance mechanism, which ensures the even distribution of data among all DataNodes.
+- Ensures metadata reliability - the transaction log mechanism is used to operate metadata, which is stored on both active and standby NameNodes. The snapshot mechanism of the file system ensures that data can be recovered promptly when a misoperation occurs.
+- Provides the security mode - HDFS provides a unique security mode to prevent a fault from spreading when a DataNode or hard disk faulty.
+
+***Data storage policy:*** HDFS supports 6 storage policies namely;
+- Hot – Storage on DISK.
+- Warm – Storage on both DISK and ARCHIVE.
+- Cold – Storage on ARCHIVE.
+- One_SSD – Storage of a single replica on SSD and other replicas on DISK.
+- All_SSD – Storage of all replicas on SSD.
+
+### 3) Configuring HDFS data storage policies
+The HDFS NameNode automatically selects DataNodes to store data replicas by default. This can be done by the following scenarios:
+
++ *Layered storage*
```

Comment:
Instead of italicizing this, you can make this as another subheading with `####` as the header